pkg/kapp/resourcesmisc/wait_rule_contract_v1.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package resourcesmisc
+
+import (
+	"fmt"
+
+	"github.com/ghodss/yaml"
+	ctlres "github.com/k14s/kapp/pkg/kapp/resources"
+	cmdtpl "github.com/k14s/ytt/pkg/cmd/template"
+	"github.com/k14s/ytt/pkg/cmd/ui"
+	"github.com/k14s/ytt/pkg/files"
+)
+
+type WaitRuleContractV1 struct {
+	ResourceMatcher ctlres.ResourceMatcher
+	Starlark        string
+}
+
+type waitRuleContractV1Result struct {
+	Result waitRuleContractV1ResultDetails
+}
+
+type waitRuleContractV1ResultDetails struct {
+	Done       bool   `json:"done"`
+	Successful bool   `json:"successful"`
+	Message    string `json:"message"`
+}
+
+func (t WaitRuleContractV1) applyYttWaitRule(res ctlres.Resource) (*waitRuleContractV1ResultDetails, error) {
+	if !t.ResourceMatcher.Matches(res) {
+		return nil, nil
+	}
+
+	return t.evalYtt(res)
+}
+
+func (t WaitRuleContractV1) evalYtt(res ctlres.Resource) (*waitRuleContractV1ResultDetails, error) {
+	opts := cmdtpl.NewOptions()
+
+	opts.DataValuesFlags.FromFiles = []string{"values.yml"}
+	opts.DataValuesFlags.ReadFileFunc = func(path string) ([]byte, error) {
+		if path != "values.yml" {
+			return nil, fmt.Errorf("Unknown file to read: %s", path)
+		}
+		return t.valuesYAML(res)
+	}
+
+	filesToProcess := []*files.File{
+		files.MustNewFileFromSource(files.NewBytesSource("resource.star", []byte(t.Starlark))),
+		files.MustNewFileFromSource(files.NewBytesSource("config.yml", t.getConfigYAML())),
+	}
+
+	out := opts.RunWithFiles(cmdtpl.Input{Files: filesToProcess}, ui.NewTTY(false))
+	if out.Err != nil {
+		return nil, fmt.Errorf("Evaluating: %s", out.Err)
+	}
+
+	if len(out.Files) == 0 {
+		fmt.Printf("Expected to find config.yml but saw zero files")
+	}
+
+	file := out.Files[0]
+	if file.RelativePath() != "config.yml" {
+		fmt.Printf("Expected config.yml but was: %s", file.RelativePath())
+	}
+
+	configObj := waitRuleContractV1Result{}
+
+	err := yaml.Unmarshal(file.Bytes(), &configObj)
+	if err != nil {
+		return nil, fmt.Errorf("Deserializing result: %s", err)
+	}
+
+	return &configObj.Result, nil
+}
+
+func (t WaitRuleContractV1) valuesYAML(res ctlres.Resource) ([]byte, error) {
```

Comment:
nit/style: recommend inlining to where its used, since it's such a small method.

Reply:
Did that earlier but just noticed that during `./hack/build.sh` it again makes it multiline. Did this as of now but whenever we run build script it will make it multiline and show it as diff.

Reply:
I believe it will be better to remain it in multiline otherwise every time we build it will show it as diff. Open for thoughts and suggestions.

Reply:
> inlining to where its used

@sethiyash I am not 100%sure, but I think what @cppforlife meant was to directly use `yaml.Marshal()` instead of `valuesYAML()` as that's the only thing it is doing.

Reply:
> I believe it will be better to remain it in multiline otherwise every time we build it will show it as diff. Open for thoughts and suggestions.

i dont understand this comment @sethiyash

Reply:
oh sorry, I thought you are saying to make this `valuesYAML()` method inline. Adopted the change to remove this method and to marshal it there directly. 

Thanks, @praveenrewar for the clarification.